### PR TITLE
upgrade from 3.9 w/o ASB to 3.10 with ASB fails

### LINKED ISSUE #8316: Upgrade from 3.9 w/o ASB to 3.10 w ASB does not work
#### Description
The migrate.yml tasks try to operate on the non existent deployment config

##### Version

```
 openshift-ansible-3.10.0-0.38.0.git.7.848b045
```

##### Steps To Reproduce
1. Install 3.9 without ASB
2. Upgrade to 3.10 with ASB

##### Expected Results
Upgrade succeeeds

##### Observed Results
```
    default: TASK [ansible_service_broker : retrieve broker configmap] **********************
    default: ok: [192.168.121.130.nip.io]
    default: 
    default: TASK [ansible_service_broker : scale down asb deploymentconfig] ****************
    default: fatal: [192.168.121.130.nip.io]: FAILED! => {"msg": "The conditional check 'broker_configmap.dao.get('type') != 'crd'' failed. The error was: error while evaluating conditional (broker_configmap.dao.get('type') != 'crd'): 'dict object' has no attribute 'data'\n\nThe error appears to have been in '/usr/share/ansible/openshift-ansible/roles/ansible_service_broker/tasks/migrate.yml': line 4, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n- block:\n    - name: scale down asb deploymentconfig\n      ^ here\n"}
    default: 
    default: TASK [ansible_service_broker : scale up asb deploymentconfig] ******************
    default: fatal: [192.168.121.130.nip.io]: FAILED! => {"msg": "The conditional check 'broker_configmap.dao.get('type') != 'crd'' failed. The error was: error while evaluating conditional (broker_configmap.dao.get('type') != 'crd'): 'dict object' has no attribute 'data'\n\nThe error appears to have been in '/usr/share/ansible/openshift-ansible/roles/ansible_service_broker/tasks/migrate.yml': line 188, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  always:\n    - name: scale up asb deploymentconfig\n      ^ here\n"}
    default: 	to retry, use: --limit @/vagrant/upgrade-3.10.retry
    default: 
    default: PLAY RECAP *********************************************************************
    default: 192.168.121.130.nip.io     : ok=511  changed=171  unreachable=0    failed=2   
    default: localhost                  : ok=31   changed=2    unreachable=0    failed=0   
    default: 
    default: 
    default: Failure summary:
    default: 
    default: 
    default:   1. Hosts:    192.168.121.130.nip.io
    default:      Play:     Upgrade Service Catalog
    default:      Task:     scale down asb deploymentconfig
    default:      Message:  The conditional check 'broker_configmap.dao.get('type') != 'crd'' failed. The error was: error while evaluating conditional (broker_configmap.dao.get('type') != 'crd'): 'dict object' has no attribute 'data'
    default:                
    default:                The error appears to have been in '/usr/share/ansible/openshift-ansible/roles/ansible_service_broker/tasks/migrate.yml': line 4, column 7, but may
    default:                be elsewhere in the file depending on the exact syntax problem.
    default:                
    default:                The offending line appears to be:
    default:                
    default:                - block:
    default:                    - name: scale down asb deploymentconfig
    default:                      ^ here
    default:                
    default: 
    default:   2. Hosts:    192.168.121.130.nip.io
    default:      Play:     Upgrade Service Catalog
    default:      Task:     scale up asb deploymentconfig
    default:      Message:  The conditional check 'broker_configmap.dao.get('type') != 'crd'' failed. The error was: error while evaluating conditional (broker_configmap.dao.get('type') != 'crd'): 'dict object' has no attribute 'data'
    default:                
    default:                The error appears to have been in '/usr/share/ansible/openshift-ansible/roles/ansible_service_broker/tasks/migrate.yml': line 188, column 7, but may
    default:                be elsewhere in the file depending on the exact syntax problem.
    default:                
    default:                The offending line appears to be:
    default:                
    default:                  always:
    default:                    - name: scale up asb deploymentconfig
    default:                      ^ here
```

##### Additional Information

If 3.9 was installed with ASB the upgrade works OK.

### PR DESCRIPTION
**Pull request by jmontleon**
Fixes #8316

### CODE REVIEW

File: roles/ansible_service_broker/tasks/upgrade.yml [hunk 1]
```diff
@@ -6,8 +6,10 @@
...
   register: broker_configmap_raw
 
+- set_fact:
+    broker_configmap: '{{ (broker_configmap_raw.results.results.0.data | from_yaml)["broker-config"] | from_yaml }}'
+  when: broker_configmap_raw.results.results.0.data is defined
+
 - name: Migrate from etcd to CRDs
   import_tasks: migrate.yml
-  when: broker_configmap.dao.get('type') != 'crd'
-  vars:
```
Comment (jmontleon) [inline roles/ansible_service_broker/tasks/upgrade.yml]: I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluate the vars before the when was evaluated so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

File: roles/ansible_service_broker/tasks/upgrade.yml [hunk 2]
```diff
@@ -6,8 +6,10 @@
...
-    broker_configmap: '{{ (broker_configmap_raw.results.results.0.data | from_yaml)["broker-config"] | from_yaml }}'
```
Comment (jmontleon) [inline roles/ansible_service_broker/tasks/upgrade.yml]: I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluating the vars before the when so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

If someone wants to take a stab at getting it to work like that feel free.

Comment (jmontleon) [inline roles/ansible_service_broker/tasks/upgrade.yml]: I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluating the vars before the when so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

If someone wants to take a stab at getting it to work like that feel free.

File: roles/ansible_service_broker/tasks/upgrade.yml [hunk 3]
```diff
@@ -6,8 +6,10 @@
...
+  when: broker_configmap is defined and broker_configmap.dao.get('type') != 'crd'
```
Comment (jmontleon) [inline roles/ansible_service_broker/tasks/upgrade.yml]: I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluating the vars before the when so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

If someone wants to take a stab at getting it to work like that feel free.

File: roles/ansible_service_broker/tasks/upgrade.yml (hunk 2 above)
Comment (jmontleon) [inline roles/ansible_service_broker/tasks/upgrade.yml]: I broke this out to a set_fact. I could not get vars: and when: on the same task to play nice. I think it was trying to evaluating the vars before the when so short circuiting with something like this wasn't working:
`when: broker_configmap_raw.results.results.0.data is defined and broker_configmap.dao.get('type') != 'crd'`

If someone wants to take a stab at getting it to work like that feel free.

**Comment by fabianvf** [comment]
Yeah, I think that will require a separate change.

**Comment by vrutkovs** [comment]
bot, retest this please

**Pull request merged**